### DISCUSSION TITLE
Code review notes...

### DISCUSSION THREAD

**Discussion by coverbeek** [frc5567/2023-Robot#194] [Team Posts]
Can we extract the Shuffleboard stuff to not have it in every class? Putting getters and setters on the individual classes, then referencing them
Alternatively, isolate Shuffleboard handling so that it isn't cluttering the robot logic.

Robot.java

Comment on shooter motor controllers what close and far mean.

Line 75, second declaration of Drivetrain
Line 80-81 setter should not be there.
Line 83-100 Should all of this be in Robot? If not, remove. If so, we need explanation.
Line 111, garbage comment
Line 139 must happen after instantiation.
Line 146 Drivetrain should be instantiated near motor controllers
Line 132 garbage comment
Line 140 Clarify comment
Line 149 Does the NetworkTable need to be passed in ? Can't it be done internally in Linelight
Line 156-157, repeat of this from 143-144
Should we move the Shuffleboard initialization code to a helper function?
Line 163, camera height should be in inches
Line 179-180 -- repeat of instantiation of distance
Line 192 is a repeat of a call in the constructor
Line 224 garbage comment
Line 244 Make the code comment clearer about what is really happening

PilotController.java

Line 18 -- comment on it's own isn't enough. Perhaps add options
Line 78 -- improve the method description to make it clear that we're scaling the values from the controller to limit the maximum output to the drivetrain.
Line 101-108 Revisit the scaling calculation (this was done at direction initially, need better mentor input)
Line 125 Both sides are the Y input (not X on the right)
Lines 110-117 Can we reduce to m_velocityScalar/m_turnScalar, and only change on gear-change?
In both arcade and tank modes, can we optimize the deadband handling so we're not copying the same code all over?
Line 161 Should we move the negation to the top of the method (Josh should know this)
Line 167 Adjust buttons to use the "Pressed" (rising edge)
Line 179 We should make note in the method description that this is called every cycle, perhaps consider making it controlDriveTrainPeriodic

Drivetrain.java

Find out if motorsafety applies to the CTRE controllers

Line 26 what is the double solenoid for? Shifting gears. Let's say so in the comment
Line 41 -- comment is wrong (Gear object not position object)
Line 54 What method is m_firstcall for?
Line 57 typo in comment, clarify what this is representing.
Line 62-69 Please do the comments and the declarations in the same order, and frankly the comments are garbage --- they tell us nothing.
Line 71-74 One comment for the two lines are sufficient
Line 76-77 These are not ever used
Line 79-80 should be removed
Line 93 Document all of the parameters
Line 95 Can we move the internal member variables to not be passed into the constructor, instead let's declare and instantiate within the drivetrain
Line 103 Comment should be emphasized and mention TEST BEFORE USE
Line 107-112 If this is really variable, we should overload this (different constructor for one solenoid vs. two solenoids)
Constructor motor calls setting configuration should move to the configDriveTrain and that should be called in the constructor
We need a no brake mode when disabled, and need to set power explicitly to 0. 
Line 134 Typo
Line 144 We should adjust so we can see all parameters (multiple lines)
159 Clean up redundant comment and we actually want to call this in the constructor.
Lines 168-173 this is redundant -- we also do this in constructor. This should probably stay and remove the constructor copy.
Lines 178 and 185 reference Quadrature encoders, but we're using the integrated encoder
Line 209-212 This is redundant with constructor. This should probably be the one place.
Line 186 - sensor should be the Integrated sensor not quad enc
Line 190 - sensor should be the Integrated sensor not quad enc 
Line 192 Not Quad encoders
Line 198, not Quad Encoders and typo
Line 226 -- Probably should be "Set" instead of "Max out"
Line 236-237 The value passed in should not be the same for both. We should have separate constants and name appropriately
Line 239, 248 FPID no no no. Danny mad. PIDF
Line 260 comment shouldn't say 10ms since the RobotMap could change
Line 296 We should only call setPistons if it is a difference from current state. Additionally we need to be sure we initialize the solenoid(s) to low gear on robot init, somehow.

Stopped at line 309

**Comment by HannahStrimpel** [comment]
Robot.java
-Delete dead code --> //victorSPX = m_intakeMotors;
-Change all the shooters to launchers so only 1 term is used
-Comment on line 114 is confusing

PilotController.java
-Should we use Getters and Setters of some kind?

Drivetrain.java
-Line 61 say what speed does
-Consider creating a scaled down value for configOpenLoopRamp since the motors are scaled
-Lines 145-148 add a comment for what each line of code is
-Delete all uses of m_counter
-Line 160 there is a typo, it should be velocity PID
-Lines 165-166 make sure motors are zeroed already, if they have been delete it at these lines because it isn't needed at this level
-Document PID more to explain why it is different for m_masterRightMotor
-Reorder PID configs
-Line 214 use robotMap constants in comment instead of 20 and 5
-During the PID config, comment if the 0s are all the same and if the 1s are all the same

**Comment by ghost** [comment]
day 2.
332 unnecessary white space
329 typo
347 should be deleted/replaced
342 typo
350 revisit first call and try to find a better way to do this 
360 remove
363 problem with returned rotate. needs to be brought in line with other code
possibly place it somewhere else, like a temporary storage for dead code. (345 to 361)
368 to 383 replace with return m_rot.at setpoint
387 add comment
388 to 421 revisit and question if it works, then go over again
412 more than one line needed
remove auton methods and re evaluate how we want to implement them
do we want an auton control class?
move 451 to 464 closer to tank drive
check arcade drive in pilot controller and compare the differences
comment on 449 is false
does getQuadraturePosition work?
does getQuadratureVelocity work?
revisit follow on 454 and 455
add ability to access gyro

launcher
address unused import on line 11
22 needs better documentation
38 better commenting about inverting
55 can be multiple lines
65 when finalizing, test motors individually to confirm inversion is correct
79 rename setMotor to setMotors
when finalized, document what positive means
95 make error a local variable, take a look at speed too
replace adjustment value everywhere to robot map
104 105 rev/revs? 
111 also has follow
119 todo for tuning?
120 unclear comment
124 if distance is greater than max distance, cap at max distance, also cap minimum
132 also has follow
130 should just be calling set velocity
side note- could put follows in their own method
142 make it private 
142 does it need a follow or does follow need to be in one place
139 clarify when it is run
149 150 add a robot map variable for inverting master
153 to 161 should be changed so the comments are more general and wont contradict future changes
160 is redundant
move 169 to robot map
168 todo for testing
167 todo for testing 
177 and down replace 0 and give that to robot map
178 todo 
191 todo
195 todo
206 and 213 potentially removed based on how we treat the adjustment value
222 remove
229 remove
all getters and setters check if we need
247- 252 should be gone
259 adjust based on above
260 multiple lines

launcher targeting - rename to limelight targeting
19 could use a shuffleboard config
46 and on 
(we want testing pid values to only be able to be changed and used, ONLY in test. end as soon as we go out of test.
pull pid values to a separate call/method/function     set
pid values reset after leaving test                              reset within disabled
stop passing in robot)
69,75,81 fix spacing issues
102 multiple lines
116 remove word once
130 is part of big change
133 typo

limelight reader
18 network table doesn't need to be passed in
23 clarify comment to explain return
26,33 make tv and tx string variables for readability 
comment like 43 on 33
44 redundant, needs to be removed
53 is redundant
56/73 get skew needs a sister function get adjusted skew to address problem in 56
56 needs better documentation for the math 
56 also needs ( )
63 white space is your friend
63 always include curly braces.
69-71 document that 0 and -90 are technically the same
remove non-implemented code
93  pi/180 could be a constant in the robot map
93  rename variables and comment better

shuffleboard shooter- change to shuffleboard launcher
38,44,all UI should have spaces, when allowed
38-47 make comments spaced evenly

Magazine- revisit after mag. is finalized
20-21 can be removed
38 clarify comment- purpose is to not let it get into the shooter
52 rename param to percent speed
62 manage stored balls needs to be re-visited when mag. is finalized
getters/setters for encoders on 100. 107, etc should be investigated if they are needed
127 replace desired with stored balls 
144 remove encoder

RobotMap
java.comments/documentation for the whole thing
25 extra space
27 velocity is misspelled
47 spacing
63 unnecessary, math.pi
65 document 4096 and replace 6 with a robotmap constant for wheel diameter
67-68 should improve documentation 
67 todo
68 # should replace 4096 on 65
63-68 should be adjusted for falcons
104-106 no longer doing 3 to 1 gear reductions, revisit.
108/109 todo
119 todo 
124 todo
140 todo
144 todo
215-218 delete
212-213 delete
234 unfinished comment
264 spacing
264 needed?

**Comment by JooshO** [comment]
![image](https://user-images.githubusercontent.com/25639881/75154900-d2f09f00-56dc-11ea-858f-c96dda6bc2ff.png)
This is the requested documentation for follow() mode, we could call it only in config but if we call a set() on the follow motors anywhere it clears the follow.
